src/modules/mc_pos_control/MulticopterPositionControl.cpp

```diff
@@ -223,17 +149,41 @@ void MulticopterPositionControl::Run()
 
 	perf_begin(_cycle_perf);
 
-	if (_local_pos_sub.update(&_local_pos)) {
+	parameters_update(false);
```

Comment:
Is this ran also when there's no subscription update to make sure the parameters are already up to date when switching to a position-controlled mode? Wouldn't it be better to only update the parameters when they are actually used?